SamplesDashboard/SamplesDashboard/Services/RepositoriesService.cs

```diff
@@ -141,23 +143,31 @@ public async Task<List<Node>> GetRepositories(string name, string endCursor = nu
         /// <returns>View count</returns>
         internal async Task<int?> FetchViews(string repoName)
         {
-            ViewData views = null;
+            var client = await CreateClient();
 
-            var httpClient = _clientFactory.CreateClient("github");                
-            HttpResponseMessage response = await httpClient.GetAsync(string.Concat("repos/microsoftgraph/", repoName, "/traffic/views"));
-            
-            if (response.IsSuccessStatusCode)
-            {
-                using (var responseStream = await response.Content.ReadAsStreamAsync())
-                using (var streamReader = new StreamReader(responseStream))
-                using (var jsonTextReader = new JsonTextReader(streamReader))
-                {
-                    var serializer = new JsonSerializer();
-                    views = serializer.Deserialize<ViewData>(jsonTextReader);                    
-                } 
-            }
-            
+            //use client to fetch views
+            var views = await client.Repository.Traffic.GetViews("microsoftgraph", repoName, new RepositoryTrafficRequest(TrafficDayOrWeek.Week));
             return views?.Count;
+        }  
+        internal async Task<GitHubClient> CreateClient()
+        {
+            var token = await _githubAuthService.GetGithubAppToken();
+
+            // Create a new GitHubClient using the installation token as authentication
+            var installationClient = new GitHubClient(new ProductHeaderValue(_config.GetValue<string>("product")))
+            {
+                Credentials = new Credentials(token)
+            };
+            return installationClient;
+        }
+
+        internal async Task<Dictionary<string, string>> FetchContributors(string repoName)
```

Comment:
The http client should be registered at the Startup.cs and then set the messagehandler.

Reply:
What do you mean by setting the messagehandler?

Reply:
Got it!